Tests/Simulation/SimulatorTestCollection.cs

```diff
@@ -32,60 +32,62 @@ public SimulatorTestCollection(IntegrationTestFixtureSim testFixture)
+        /// However all the method calls where non existing
+        /// </summary>
+        /// <returns></returns>
+        [Fact]
+        public async Task Ten_Devices_Sending_Messages_At_Same_Time()
+        {
+            var networkServerIPEndpoint = CreateNetworkServerEndpoint();
+            using (var simulatedBasicsStation = new SimulatedBasicsStation(StationEui.Parse(StationEUI), networkServerIPEndpoint))
+            {
+                await simulatedBasicsStation.StartAsync();
+
+                var deviceTasks = new List<Task>();
+                foreach (var device in TestFixtureSim.DeviceRange1000_ABP)
+                {
+                    var simulatedDevice = new SimulatedDevice(device);
+                    using var request = WaitableLoRaRequest.CreateWaitableRequest(simulatedDevice.CreateConfirmedDataUpMessage(device.DeviceID));
+
+                    deviceTasks.Add(simulatedBasicsStation.SendDataMessageAsync(request));
+                    await Task.Delay(2000);
+                }
+
+                await Task.WhenAll(deviceTasks);
+                await simulatedBasicsStation.StopAsync();
+            }
+
+            var eventsByDevices = TestFixture.IoTHubMessages.Events.GroupBy(x => x.SystemProperties["iothub-connection-device-id"]);
+            // There are 11 devices in that group 0-10
+            Assert.Equal(11, eventsByDevices.Count());
+        }
         [Fact]
         public async Task Single_ABP_Simulated_Device()
         {
             const int MessageCount = 5;
 
             var device = TestFixtureSim.Device1001_Simulated_ABP;
-            var simulatedDevice = new SimulatedDevice(device);
+            var simulatedDevice = new Common.SimulatedDevice(device);
             var networkServerIPEndpoint = CreateNetworkServerEndpoint();
 
-            using var simulatedPacketForwarder = new SimulatedPacketForwarder(networkServerIPEndpoint);
-            simulatedPacketForwarder.Start();
+            using var simulatedBasicsStation = new SimulatedBasicsStation(StationEui.Parse(StationEUI), networkServerIPEndpoint);
+            await simulatedBasicsStation.StartAsync();
 
             for (var i = 1; i <= MessageCount; i++)
             {
-                await simulatedDevice.SendUnconfirmedMessageAsync(simulatedPacketForwarder, i.ToString(CultureInfo.InvariantCulture));
-                // await simulatedDevice.SendConfirmedMessageAsync(simulatedPacketForwarder, i.ToString());
+                using var request = WaitableLoRaRequest.CreateWaitableRequest(simulatedDevice.CreateConfirmedDataUpMessage(i.ToString(CultureInfo.InvariantCulture)));
+                await simulatedBasicsStation.SendDataMessageAsync(request);
                 await Task.Delay(this.intervalBetweenMessages);
             }
+            var msgsFromDevice = TestFixture.IoTHubMessages.Events.Where(x => x.GetDeviceId() == simulatedDevice.LoRaDevice.DeviceID);
```

Comment:
[nit] instead of filtering and then counting, we can count directly

```csharp
var actualAmountOfMsgs = TestFixture.IoTHubMessages.Events.Count(x => x.GetDeviceId() == simulatedDevice.LoRaDevice.DeviceID && x.Properties.ContainsKey("iothub-message-schema"));
```